src/vec_vector.h

```diff
@@ -68,9 +68,15 @@ template <typename T>
 class Vector {
   public:
     const gf::RingModN<T>* rn;
+
     Vector(const gf::RingModN<T>& rn, int n, T* mem = nullptr, int mem_len = 0);
     Vector(const gf::RingModN<T>& rn, std::initializer_list<T> values);
+    Vector(const Vector& other);
+    Vector(Vector&& other) noexcept;
```

Comment:
I see that these two new constructors are used only for fft_utest.cp. I think it's better to fix the issue in the tests instead of modifying vec_vector.h

Reply:
The thing is, there is no problem with the tests but there is a problem of bad design with the Vector class here.

The Vector class owns a resource, and to properly handle this we have provided a user-defined destructor. Once you do that, you have to follow the [Rule of Three](https://en.cppreference.com/w/cpp/language/rule_of_three) (Rule of Five since C++11).

In general, when you have a class that manage some resources you have the choice to consider you class either as:
- a value-semantic (that's the case for vector): you have to follow the Rule of 3/5
- an entity-semantic: you should forbid the copy (and maybe implement the moves).

The fix you suggest would be to use `=` as before but that's bad for two reasons:
- this doesn't fix the flaw in the class's design
- using the copy assignment instead of the copy constructor is slower (not like we care in the tests, but that's still ugly)

Reply:
Oh you're right 👍